File: src/slagalica/core/ReverseStringComparator.java
package slagalica.core;

import java.util.Comparator;

public class ReverseStringComparator implements Comparator<String>{
	@Override
	public int compare(String first, String second) {
		if (first.length() < second.length()) {
			return 1;
		} else if (first.length() > second.length()) {
			return -1;
		}
		return -1 * first.compareTo(second);
	}
}

File: src/main/java/slagalica/core/ReverseStringComparator.kt
package slagalica.core

class ReverseStringComparator : Comparator<String> {
    override fun compare(first: String, second: String): Int = when {
        first.length < second.length -> 1
        first.length > second.length -> -1
        else -> -1 * first.compareTo(second)
    }
}